cmake/setup_compiler_flags_gnu.cmake

```diff
@@ -187,6 +187,12 @@ IF (CMAKE_BUILD_TYPE MATCHES "Release")
   ENABLE_IF_SUPPORTED(DEAL_II_CXX_FLAGS_RELEASE "-funroll-all-loops")
   ENABLE_IF_SUPPORTED(DEAL_II_CXX_FLAGS_RELEASE "-fstrict-aliasing")
 
+  #
+  # Disable assertions in the standard library in release mode
```

Comment:
Would you mind changing this to `Disable assert() in deal.II and user projects in release mode`?
Because, technically speaking, `-DNDEBUG` is not controlling assertions in the standard library itself.